### PR TITLE
[DCJ-222] Refactor DatasetRegistration.js to JSX

### PR DESCRIPTION
**Pull request by PintoGideon**
### Addresses

https://broadworkbench.atlassian.net/browse/DCJ-222

### Summary

1. Remove the use of react-hyperscript-helpers
2. Update the file extension from js to jsx
3. _This component needs to be rigorously tested_
4. This component breaks in production due to an earlier refactor to the ConfirmationDialog.jsx component. This PR should fix it.

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Comment by fboulnois** [comment]
I rebased on develop to include the latest JSX changes, I didn't make any other changes.

File: src/pages/DatasetRegistration.jsx [hunk 1]
```diff
@@ -0,0 +1,1461 @@
...
+  };
+};
+
+const controlLabelStyle = {
+  fontWeight: 500,
+  marginTop: '1rem',
+  marginBottom: '1rem'
+};
+
+const DatasetRegistration = (props) => {
```
Comment (rushtong) [inline src/pages/DatasetRegistration.jsx:79]: Can we retain this comment?
```
// TODO: Deprecated - remove this component when all datasets have been converted to studies
```
This component does need to be removed. In production, all datasets have been converted to studies, so this is essentially a relic for development. Alternatively, we can ticket this for removal.

File: src/pages/DatasetRegistration.jsx [hunk 2]
```diff
@@ -0,0 +1,1461 @@
...
+                          id='checkMethods'
+                          type='checkbox'
+                          className='checkbox-inline rp-checkbox'
+                          name='methods'
+                          disabled={isUpdateDataset}
+                        />
+                        <label className='regular-checkbox rp-choice-questions' htmlFor='checkMethods'>
+                          <span className='access-color'>No methods development or validation studies (NMDS)</span>
+                        </label>
+                      </div>
```
Comment (rushtong) [inline src/pages/DatasetRegistration.jsx:1180]: There is a slight nested div issue here. The checkboxes beneath this one are indented. These two blocks should look like this:
```
                    <div className='form-group'>
                      <div className='col-lg-12 col-md-12 col-sm-12 col-xs-12 rp-group'>
                        <label className='control-label rp-title-question dataset-color'>
                           2.2 Secondary Data Use Terms
                          <span>Please select all applicable data use parameters.</span>
                        </label>
                      </div>
                      <div className='col-lg-12 col-md-12 col-sm-12 col-xs-12 rp-group'>
                        <div className='checkbox'>
                          <input
                            checked={methods}
                            onChange={handleCheckboxChange}
                            id='checkMethods'
                            type='checkbox'
                            className='checkbox-inline rp-checkbox'
                            name='methods'
                            disabled={isUpdateDataset}
                          />
                          <label className='regular-checkbox rp-choice-questions' htmlFor='checkMethods'>
                            <span className='access-color'>No methods development or validation studies (NMDS)</span>
                          </label>
                        </div>
                      </div>
```

**Review by rushtong — APPROVED** [review]
Reviewed locally, very minor issues - address as you see fit. This page in dev/staging/prod is non-functional:
```
Uncaught Error: Invalid hook call. Hooks can only be called inside of the body of a function component. 
``` 
So, any changes to this that fix the rendering is 👍🏽 
Please file a ticket to remove this page altogether.

File: src/pages/DatasetRegistration.jsx [hunk 3]
```diff
@@ -0,0 +1,1461 @@
...
+
+  const handleChange = (e) => {
+    const field = e.target.name;
+    const value = e.target.value;
+    setRegistrationState(prevState => ({
+      ...prevState,
+      datasetData: {
+        ...prevState.datasetData,
+        [field]: value
+      },
```
Comment (fboulnois) [inline src/pages/DatasetRegistration.jsx]:
```suggestion
      datasetData[field]: value
```

Would this work instead? The current syntax looks a bit confusing.

File: src/pages/DatasetRegistration.jsx [hunk 4]
```diff
@@ -0,0 +1,1461 @@
...
+  const handlePositiveIntegerOnly = (e) => {
+    const field = e.target.name;
+    const value = e.target.value.replace(/[^\d]/, '');
+
+    if (value === '' || parseInt(value, 10) > -1) {
+      setRegistrationState(prevState => ({
+        ...prevState,
+        datasetData: {
+          ...prevState.datasetData,
+          [field]: value
```
Comment (fboulnois) [inline src/pages/DatasetRegistration.jsx]: Ditto here.

File: src/pages/DatasetRegistration.jsx [hunk 5]
```diff
@@ -0,0 +1,1461 @@
...
+  };
+
+  const handleCheckboxChange = (e) => {
+    const field = e.target.name;
+    const value = e.target.checked;
+    setRegistrationState(prevState => ({
+      ...prevState,
+      formData: {
+        ...prevState.formData,
+        [field]: value
```
Comment (fboulnois) [inline src/pages/DatasetRegistration.jsx]: Ditto here.

File: src/pages/DatasetRegistration.jsx [hunk 6]
```diff
@@ -0,0 +1,1461 @@
...
+      return registrationState.datasetData.isValidName ? 'form-control' : 'form-control required-field-error';
+    }
+  };
+
+  const attestAndSave = () => {
+    setRegistrationState(prevState => {
+      let allValid = validateRequiredFields(prevState.datasetData);
+      return {
+        ...prevState,
+        showDialogSubmit: allValid ? true : false,
```
Comment (fboulnois) [inline src/pages/DatasetRegistration.jsx]:
```suggestion
        showDialogSubmit: allValid,
```

File: src/pages/DatasetRegistration.jsx [hunk 7]
```diff
@@ -0,0 +1,1461 @@
...
+        problemLoadingUpdateDataset: false,
+        showValidationMessages: allValid ? false :true
```
Comment (fboulnois) [inline src/pages/DatasetRegistration.jsx]:
```suggestion
        showValidationMessages: !allValid
```

File: src/pages/DatasetRegistration.jsx (hunk 1 above)
Comment (PintoGideon) [reply src/pages/DatasetRegistration.jsx:79]: Thanks, will do @rushtong

**Review by fboulnois — APPROVED** [review]
A few suggestions, otherwise lgtm:

File: src/pages/DatasetRegistration.jsx (hunk 3 above)
Comment (PintoGideon) [reply src/pages/DatasetRegistration.jsx]: Agreed. I made an update to this.

**Pull request merged**